### ISSUE TITLE
Software bill of materials (SBOM) support

### ISSUE THREAD

**Issue by OskarsPakers** [go-gitea/gitea#25440]
### Feature Description

It would be useful to upload and view dependencies of a repo using software bill of materials manifests.
Similar feature exists in Github https://github.blog/2023-03-28-introducing-self-service-sboms/

### Screenshots

![image](https://github.com/go-gitea/gitea/assets/3343347/b439a645-f706-40eb-9114-4d8eea1ce913)

**Comment by silverwind** [comment]
What libraries does GitHub use to implement this? Are they Ruby or something else?

**Comment by OskarsPakers** [comment]
I dont know what Github uses but I have generated sbom with these
https://github.com/microsoft/sbom-tool (C#)
https://github.com/anchore/syft (Go)

Syft looks like a good choice.

**Comment by silverwind** [comment]
Yes, syft looks promising.

**Comment by lunny** [comment]
Maybe it should be a part of a release, a tag, or a package? Output development version SBOM sounds unusable.

**Comment by OskarsPakers** [comment]
I think it depends on the branching strategy. If it is trunk-based development, where each commit is a release candidate, I would want to have SBOM for each commit.
I think extracting SBOM data as such should be part of the pipeline, but storing and viewing the dependency data would be needed in Gitea. Perhaps within packages section.

**Comment by ggjulio** [comment]
> I think extracting SBOM data as such should be part of the pipeline, but storing and viewing the dependency data would be needed in Gitea

agree, some sboms are non trivial and includes image scanning, not just code.
Giving an sbom to ingest would be nice.

> If we can make it selectable for branch/commit/tag, I guess it would be nice to have, but not required.

Would be very nice too, github is not able to do that, which don't help for tracking new vuln in releases.
To go further, on the releases (which are tags) it would be nice to have a small summary of critical, high, low vuln included in the release.

<img width="1919" height="253" alt="Image" src="https://github.com/user-attachments/assets/96f0f92a-b29f-4988-bb6b-2eaae18302a1" />
